app/models/company.rb

```diff
@@ -1,5 +1,24 @@
 class Company < ActiveRecord::Base
   validates :name, :city, presence: true
   validates :name, uniqueness: true
-  has_many :jobs
+  has_many :jobs, dependent: :destroy
+  has_many :contacts
+
+  def self.top_three_companies_by_interest
+    top_3 = joins(:jobs).group(:id).order("AVG(jobs.level_of_interest) DESC").limit(3).pluck(:name, "AVG(jobs.level_of_interest)")
+    top_3.map {|company| [company[0], company[1].to_f]}
```

Comment:
this could be a separate method because you've already found the top three companies. this method is simply formatting it in a way that's going to be easier to use!